### PR TITLE
[BAHIR-18] include examples as test sources in Maven build

### PR DESCRIPTION
**Pull request by ckadner**
[BAHIR-18: Include examples in Maven test build](https://issues.apache.org/jira/browse/BAHIR-18) 

This PR is to include the examples in the Maven build configuration. By defining them as _"additional test sources"_ we make sure ...
- the examples get compiled
- IDEs like IntelliJ or Eclipse recognize the `<module>/examples/src/[java|scala]` as source folders
- keep the examples along with their additional dependencies excluded from the binaries

### CODE REVIEW

**Comment by lresende** [comment]
This did not trigger the jira integration, as it has the wrong JIRA number, could you please correct that on your title and commit.

**Comment by lresende** [comment]
Adding a comment to trigger JIRA integration

File: pom.xml [hunk 1]
```diff
@@ -867,6 +873,26 @@
...
+            </goals>
+            <configuration>
+              <sources>
+                <source>${basedir}/examples/src/main/java</source>
+                <source>${basedir}/examples/src/main/scala</source>
+              </sources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
```
Comment (lresende) [inline pom.xml:895]: @ckadner you might want to check if you need to enable scalastyle for the examples source folders right below....

**Comment by ckadner** [comment]
@lresende I pushed your suggested update. Good catch!

**Pull request closed without merging**